foreshadow/preparer.py

```diff
@@ -107,3 +121,90 @@ def _get_params(self, attr, deep=True):
         out.update({"steps": steps})  # manually
         # adding steps to the get_params()
         return out
+
+    # def dict_serialize(self, deep=False):
+    #     """Serialize the init parameters (dictionary form) of a pipeline.
+    #
+    #     This method removes redundant column_sharers in the individual
+    #     steps.
+    #
+    #     Note:
+    #         This recursively serializes the individual steps to facilitate a
+    #         human readable form.
+    #
+    #     Args:
+    #         deep (bool): If True, will return the parameters for this estimator
+    #             recursively
+    #
+    #     Returns:
+    #         dict: The initialization parameters of the pipeline.
+    #
+    #     """
+    #     serialized = super().dict_serialize(deep=deep)
+    #     column_sharer_serialized = serialized.pop("column_sharer", None)
+    #     # Remove all instance of column_sharer from the serialized recursively.
+    #     serialized = self.__remove_key_from(serialized, target="column_sharer")
+    #     # Add back the column_sharer in the end only once.
+    #     serialized["column_sharer"] = column_sharer_serialized
+    #     return serialized
+
+    def dict_serialize(self, deep=True):
+        import pdb
+
+        pdb.set_trace()
+        params = self.get_params(deep=deep)
+        selected_params = self.__create_selected_params(params)
+        serialized = _make_serializable(
+            selected_params, serialize_args=self.serialize_params
+        )
+        column_sharer_serialized = serialized.pop("column_sharer", None)
+        serialized = self.__remove_key_from(serialized, target="column_sharer")
+        # Add back the column_sharer in the end only once.
+        serialized["column_sharer"] = column_sharer_serialized
+        steps = serialized["steps"]
+        steps_reformatted = [{step[0]: step[1]} for step in steps]
+        serialized["steps"] = steps_reformatted
+        return serialized
+
+    @classmethod
+    def dict_deserialize(cls, data):
+        params = _make_deserializable(data)
+        params["steps"] = [list(step.items())[0] for step in params["steps"]]
+        deserialized = cls(**params)
+
+        deserialized.configure_column_sharer(deserialized.column_sharer)
+
+        return deserialized
+
+    def configure_column_sharer(self, column_sharer):
+        for step in self.steps:
+            step[1].configure_column_sharer(column_sharer)
+
+    def __create_selected_params(self, params):
+        init_params = inspect.signature(self.__init__).parameters
+        selected_params = {
+            name: params.pop(name)
+            for name in init_params
+            if name not in ["self", "kwargs"]
+        }
+        selected_params["steps"] = params.pop("steps")
+        return selected_params
+
+    def __remove_key_from(self, data, target="column_sharer"):
```

Comment:
Can we document this function and make it a private method I see this being useful.